File: ios/Classes/PluginAnalyticsPlugin.swift
import Flutter

public class PluginAnalyticsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
      let api = PianoAnalyticsApiImpl()
      PianoAnalyticsApiSetup.setUp(binaryMessenger: registrar.messenger(), api: api)
  }
}

File: android/src/main/kotlin/com/zenika/plugin_analytics/PluginAnalyticsPlugin.kt
package com.zenika.plugin_analytics

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** PluginAnalyticsPlugin */
class PluginAnalyticsPlugin: FlutterPlugin {

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    PianoAnalyticsApi.setUp(flutterPluginBinding.binaryMessenger, PianoAnalyticsApiImpl(flutterPluginBinding.applicationContext))
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
  }
}
